test/mpi/impls/mpich/threads/pt2pt/sendrecv_vci_assignment.c

```diff
@@ -0,0 +1,199 @@
+                }
+
+                if (strcmp(hint_buf, comm_hints_vals[j])) {
+                    fprintf(stderr,
+                            "Error while reading value of hint : %s, value read: %s instead of: %s\n",
+                            comm_hints[j], hint_buf, comm_hints_vals[j]);
+                    errs++;
+                }
+            }
+            MPI_Info_free(&comm_info_out);
+        }
+    }
+
+    /* Setup and launch the test threads */
+    for (i = 0; i < NTHREADS; i++) {
+        params[i].comm = dup_world;
+        params[i].tag = i;      /* Set tag = thread id */
+        params[i].iter = ITERATIONS;
+        params[i].id = i;
+        if (i == NTHREADS - 1)
+            run_send_recv_test(&params[i]);
+        else
+            MTest_Start_thread(run_send_recv_test, &params[i]);
+    }
+    MTest_Join_threads();
+
+    /* Cleanup comm objects */
+    if (n_hints > 0)
+        MPI_Info_free(&comm_info);
+    MPI_Comm_free(&dup_world);
+
+    for (i = 0; i < NTHREADS; i++)
+        errs += params[i].result;
+
+    return errs;
+}
+
+int main(int argc, char **argv)
+{
+    int i, pmode, nprocs, errs = 0, err, count = 0, iter = 0;
+    const char *hints_1[2];
+    const char *hints_1_vals[2];
+
+    MTest_Init_thread(&argc, &argv, MPI_THREAD_MULTIPLE, &pmode);
+    if (pmode != MPI_THREAD_MULTIPLE) {
+        fprintf(stderr, "MPI_THREAD_MULTIPLE not supported by the MPI implementation\n");
+        MPI_Abort(MPI_COMM_WORLD, 1);
+    }
+
+    MPI_Comm_size(MPI_COMM_WORLD, &nprocs);
+    if (nprocs < 2) {
+        fprintf(stdout, "Need two processes\n");
+        MPI_Abort(MPI_COMM_WORLD, 1);
+    }
+
+    err = MTest_thread_barrier_init();
+    if (err) {
+        fprintf(stderr, "Could not create thread barrier\n");
+        MPI_Abort(MPI_COMM_WORLD, 1);
+    }
+
+    MPI_Barrier(MPI_COMM_WORLD);
+
+    /* Test : no hint
+     * Perform send/recv with no comm hint applied. Just for
+     * sanity check.
+     */
+    errs += comm_hint_test(NULL, NULL, 0, false);
+    MPI_Barrier(MPI_COMM_WORLD);
+
+    /* TODO: Standard comm hints tests. */
+
+    /* Non-standard comm hints tests.
+     * The following non-standard hints allow user to directly
+     * suggest what vci_idx to use for sender and receiver. Again, with
+     * these hints, we expect send-recv to run successfully.
+     */
+
+    /* Test : hints sender_vci and receiver_vci with valid values.
+     * Here we expect the program to run successfully when these valid
+     * values are provided.
+     */
+    hints_1[0] = "sender_vci";
+    hints_1[1] = "receiver_vci";
+    hints_1_vals[0] = "2";
+    hints_1_vals[1] = "3";
+    errs += comm_hint_test(hints_1, hints_1_vals, 2, false);
+    MPI_Barrier(MPI_COMM_WORLD);
+
+    /* Test : sender_vci and receiver_vci with potential invalid values
+     * Here we try to pass invalid values for sender and reveiver vci_id.
+     * We expect MPI to ignore these values and use internal vci_idx selection
```

Comment:
Internally mod will be applied, so I don't think it is accurate to say it will be ignored.